src/gluonnlp/models/bart.py

```diff
@@ -0,0 +1,388 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+BART Model
+
+@article{lewis2019bart,
+    title = {BART: Denoising Sequence-to-Sequence Pre-training for Natural
+Language Generation, Translation, and Comprehension},
+    author = {Mike Lewis and Yinhan Liu and Naman Goyal and Marjan Ghazvininejad and
+              Abdelrahman Mohamed and Omer Levy and Veselin Stoyanov
+              and Luke Zettlemoyer },
+    journal={arXiv preprint arXiv:1910.13461},
+    year = {2019},
+}
+
+"""
+
+__all__ = ['BartModel', 'list_pretrained_bart', 'get_pretrained_bart']
+
+import os
+from typing import Tuple
+
+import mxnet as mx
+from mxnet import use_np
+from mxnet.gluon import nn
+
+from ..base import get_model_zoo_home_dir, get_repo_model_zoo_url, \
+                   get_model_zoo_checksum_dir
+from ..registry import BACKBONE_REGISTRY
+from ..utils.misc import download, load_checksum_stats
+from .transformer import TransformerModel
+from ..utils.config import CfgNode as CN
+from ..utils.registry import Registry
+from ..data.tokenizers import HuggingFaceByteBPETokenizer
+
+bart_cfg_reg = Registry('bart_cfg')
+
+
+@bart_cfg_reg.register()
+def bart_base():
+    cfg = CN()
+    # Config for the bart base model
+    cfg.MODEL = CN()
+    cfg.MODEL.vocab_size = 51201
+    cfg.MODEL.max_src_length = 1024
+    cfg.MODEL.max_tgt_length = 1024
+    cfg.MODEL.scale_embed = False
+    cfg.MODEL.pos_embed_type = 'learned'
+    cfg.MODEL.shared_embed = True
+    cfg.MODEL.tie_weights = True
+    cfg.MODEL.attention_dropout = 0.1
+    cfg.MODEL.activation_dropout = 0.0
+    cfg.MODEL.dropout = 0.1
+    cfg.MODEL.layer_norm_eps = 1E-5
+    cfg.MODEL.pooler_activation = 'tanh'
+    cfg.MODEL.data_norm = True
+    cfg.MODEL.layout = 'NT'
+    cfg.MODEL.dtype = 'float32'
+
+    # Parameters for the encoder
+    cfg.MODEL.ENCODER = CN()
+    cfg.MODEL.ENCODER.num_layers = 6
+    cfg.MODEL.ENCODER.units = 768
+    cfg.MODEL.ENCODER.num_heads = 12
+    cfg.MODEL.ENCODER.hidden_size = 3072
+    cfg.MODEL.ENCODER.recurrent = False
+    cfg.MODEL.ENCODER.pre_norm = False
+    cfg.MODEL.ENCODER.activation = 'gelu'
+    cfg.MODEL.ENCODER.use_qkv_bias = True
+
+    # Parameters for the decoder
+    cfg.MODEL.DECODER = CN()
+    cfg.MODEL.DECODER.num_layers = 6
+    cfg.MODEL.DECODER.units = 768
+    cfg.MODEL.DECODER.num_heads = 12
+    cfg.MODEL.DECODER.hidden_size = 3072
+    cfg.MODEL.DECODER.recurrent = False
+    cfg.MODEL.DECODER.pre_norm = False
+    cfg.MODEL.DECODER.activation = 'gelu'
+    cfg.MODEL.DECODER.use_qkv_bias = True
+
+    # Parameters for the initializer
+    cfg.INITIALIZER = CN()
+    cfg.INITIALIZER.embed = ['normal', 768**-0.5]
```

Comment:
Use `['xavier', 'gaussian', 'in', 1.0]`, you may check the implementation in https://mxnet.apache.org/versions/1.6/api/python/docs/_modules/mxnet/initializer.html#Xavier.